### PR TITLE
[iOS] - Memory Debugger Stub

### PR DESCRIPTION
**Pull request by Brandon-T**
Stub out memory debugger to prevent _performMemoryWarning private-api enabled via CHROMIUM_BRANDING

Resolves https://github.com/brave/brave-browser/issues/14119

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)
- [ ] Requested a security/privacy review as needed

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by bsclifton — APPROVED** [review]
Was originally done in 1.15 and 1.16 with https://github.com/brave/brave-core/pull/6866/commits/1801902f60d5ec71c135216264be2727d3e9187c - we forgot to pull into master

Change LGTM 👍

**Comment by bsclifton** [comment]
ping @bridiver or @mkarolin for patch review 😄

**Comment by mkarolin** [comment]
Do we not care about `MemoryDebugger` in our builds at all? If it's just the `CHROMIUM_BRANDING` parts that we want to avoid, we can always disable just those via a #define. If we don't need it at all in our builds, seems like there are some header #includes that aren't needed?

**Comment by kylehickinson** [comment]
We don't care about MemoryDebugger at all, this code is unfortunately compiled in due to Chromium's iOS sync implementation being wired to UI dependencies. We don't show or use any of their UI

**Comment by mkarolin** [comment]
Ok, then maybe we could clean up those #includes that aren't needed.

**Comment by bsclifton** [comment]
Failed test not related to this; ready for re-review @mkarolin 😄 
```
1 test failed:

[2021-02-13T01:02:02.114Z]     BraveNavigatorUserAgentFarblingBrowserTest.FarbleNavigatorUserAgent (../../brave/browser/farbling/brave_navigator_useragent_farbling_browsertest.cc:138)
```

File: chromium_src/ios/chrome/browser/memory/memory_debugger.mm [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#import "ios/chrome/browser/memory/memory_debugger.h"
+#include <stdint.h>
+#include <memory>
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+namespace {
+}  // namespace
```
Comment (mkarolin) [inline chromium_src/ios/chrome/browser/memory/memory_debugger.mm:14]: Is this needed?

**Review by mkarolin — APPROVED** [review]
++

File: chromium_src/ios/chrome/browser/memory/memory_debugger.mm (hunk 1 above)
Comment (bsclifton) [reply chromium_src/ios/chrome/browser/memory/memory_debugger.mm:14]: Nope - will remove 👍

**Pull request merged**